core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.spec.ts

```diff
@@ -311,11 +313,36 @@ describe('Contribution and review service', () => {
           '2022-01-02'
         )
         .then(response => {
-          expect(response.from_date).toEqual('1 Nov 2022');
-          expect(response.to_date).toEqual('1 Dec 2022');
-          expect(response.contribution_hours).toEqual(1.0);
-          expect(response.team_lead).toEqual('Test User');
-          expect(response.language).toEqual('Hindi');
+          const data = response.certificate_data;
+          expect(data).toBeDefined();
+          expect(data?.from_date).toEqual('1 Nov 2022');
```

Comment:
Why do you need the question marks here and below?

Reply:
If data == null, then the lines `expect(data.from_date).toEqual('1 Nov 2022')` and below give a type error.

I was considering adding the lines below, which would satisfy the type checker for the expect expressions, but I later went with the other approach.
```
if (!data) {
     fail("some fail message here");
     return;
}
```

I couldn't find another example of this situation in the code, so it was hard to tell what would be best stylistically. Please give me your thoughts!

Reply:
I changed the test to use the `if (!data) ... ` block to satisfy the type checker. There might be a different way you'd want an optional value such as this to be handled within the test. Just let me know.